sys/src/9/ufs/ufs/ufs_vnops.c

```diff
@@ -2394,28 +2407,31 @@ ufs_pathconf (struct vop_pathconf_args *ap)
 	return (error);
 }
 
+#endif // 0
+
 /*
  * Initialize the vnode associated with a new inode, handle aliased
  * vnodes.
  */
 int 
-ufs_vinit (struct mount *mntp, struct vop_vector *fifoops, struct vnode **vpp)
+ufs_vinit (MountPoint *mntp, vnode **vpp)
 {
-	struct inode *ip;
-	struct vnode *vp;
+	inode *ip;
+	vnode *vp;
 
 	vp = *vpp;
 	ip = VTOI(vp);
-	vp->v_type = IFTOVT(ip->i_mode);
-	if (vp->v_type == VFIFO)
-		vp->v_op = fifoops;
-	ASSERT_VOP_LOCKED(vp, "ufs_vinit");
+	vp->v_type = ifmt_to_vtype(ip->i_mode);
+	// TODO HARVEY Locking
+	//ASSERT_VOP_LOCKED(vp, "ufs_vinit");
```

Comment:
It needs to be a lock that allows recursion.  For some reason I though qlocks don't, but looking at the code, it seems they'd be fine.